content/docs/getting-started/hello-world.md

```diff
@@ -175,27 +141,61 @@ different pros and cons. In this example, we will use the default executor of th
 [Tokio runtime][rt].
 
 ```rust
-# #![deny(deprecated)]
-# extern crate tokio;
-# extern crate futures;
-#
-# use tokio::prelude::*;
-# use futures::future;
-# fn main() {
-# let client = future::ok(());
-println!("About to create the stream and write to it...");
-tokio::run(client);
-println!("Stream has been created and written to.");
-# }
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpStream;
+use tokio::runtime::Runtime;
+
+fn main() {
+    let addr = "127.0.0.1:6142";
+    let client = TcpStream::connect(addr);
+
+    let future = async move {
+        if let Ok(mut stream) = client.await {
+            println!("created stream");
+
+            let result = stream.write_all(b"hello world\n").await;
+
+            println!("wrote to stream; success={:?}", result.is_ok());
+        }
+    };
+
+    let rt = Runtime::new().unwrap();
+    println!("About to create the stream and write to it...");
+    rt.block_on(future);
+    println!("Stream has been created and written to.");
+}
 ```
 
-`tokio::run` starts the runtime, blocking the current thread until all spawned tasks
+In this case we create the default runtime, and spawn the task onto it, blocking the current thread until all spawned tasks
 have completed and all resources (like files and sockets) have been dropped.
 
 So far, we only have a single task running on the executor, so the `client` task
 is the only one blocking `run` from returning. Once `run` has returned we can be sure
 that our Future has been run to completion.
 
+This example can also be greatly simplified using the `#[tokio::main]` procedural macro.
+
+```rust
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpStream;
+
+#[tokio::main]
+async fn main() {
+    let addr = "127.0.0.1:6142";
+    let client = TcpStream::connect(addr);
+
+    if let Ok(mut stream) = client.await {
+        println!("created stream");
+
+        let result = stream.write_all(b"hello world\n").await;
+
+        println!("wrote to stream; success={:?}", result.is_ok());
+    }
+}
+```
```

Comment:
Ah, I see we introduce `#[tokio::main]` here...I still think it _might_ be better to show the easier API first, and then say "okay, here is what `tokio::main` actually _does_" and show code that manually creates a runtime.